GLMakie/assets/shader/util.vert

```diff
@@ -239,30 +239,41 @@ out vec3 o_view_pos;
 out vec3 o_normal;
 out vec3 o_lightdir;
 out vec3 o_camdir;
+
 // transpose(inv(view * model))
 // Transformation for vectors (rather than points)
 uniform mat3 normalmatrix;
 uniform vec3 lightposition;
 uniform vec3 eyeposition;
 uniform float depth_shift;
 
-
+// TODO: maybe move some of this to lighting.vert or vice versa?
 void render(vec4 position_world, vec3 normal, mat4 view, mat4 projection, vec3 lightposition)
 {
     // normal in world space
     o_normal = normalmatrix * normal;
+
     // position in view space (as seen from camera)
     vec4 view_pos = view * position_world;
+    view_pos /= view_pos.w;
+
     // position in clip space (w/ depth)
     gl_Position = projection * view_pos;
     gl_Position.z += gl_Position.w * depth_shift;
-    // direction to light
-    o_lightdir = normalize(view*vec4(lightposition, 1.0) - view_pos).xyz;
+
+    // direction to light for :fast shading
+    // (normalizing this will result in incorrect lighting)
+    vec4 view_light_pos = view*vec4(lightposition, 1.0);
+    o_lightdir = view_light_pos.xyz / view_light_pos.w - view_pos.xyz;
```

Comment:
As noted normalizing here results in incorrect light directions/perceived positions. To illustrate this, consider a light source shining on a rect viewed from the side:

![Screenshot from 2023-09-27 16-27-14](https://github.com/MakieOrg/Makie.jl/assets/10947937/7739c765-f5a4-4154-8cf7-6e7fa2e6c7ac)

If we **don't** normalize `o_lightdir` it will be a vector pointing from the light source to the vertex handled by the vertex shader. Since I placed the light source at (0, 0) here they'll match p1 and p2 in the example. On the way to the fragment shader `o_lightdir` will then get interpolated between different vertices. For example, halfway between the two vertices the vector will point to the  green "Vertex center". Normalizing in the fragment shader will then give correct light-to-vertex-center direction.

If we **do** normalize `o_lightdir` in the vertex shader as before, we will be interpolating the normalized `o_lightdir` vectors. In the sketch above those point to n1 (equal to p1) and n2. Doing the same half-way interpolation for vertices as above, we end up with a `o_lightdir` halfway between n1 and n2, which is marked as "Normalized center". This is not a vector passing through the vertex center position and thus the wrong light direction.

I'm pretty sure the test failure in `Order Independent transparency` is a result of this change. There are likely quite a few more failures that didn't get caught.

Reply:
I switched to doing lighting in world space so this is not really relevant to lightdir anymore. Though it probably still applies to camdir. (World vs view/camera space shouldn't matter for lighting. It was using view/camera space because SSAO needed it and lighting could reuse it. Merging lighting between volume and the rest of Makie requires world space though.)